tests/jballerina-unit-test/src/test/resources/test-src/record/map_to_record.bal

```diff
@@ -177,6 +177,23 @@ function testComplexSubtyping2() {
     gR.x = rC;
 }
 
+json jsonMapping = {a : 1};
+
+function testImmutableJsonMappingToExclusiveRecord() {
+    json readonlyJsonMapping = jsonMapping.cloneReadOnly();
+    Bar intBar = <Bar>readonlyJsonMapping;
+    assert(1, intBar.a);
```

Comment:
Shall we add negative runtime tests also? 

```
Bar|error res = trap <Bar>mapping; // and assert error
```

Reply:
Please check now 751bc346250f938df72c4a123829017d1a9aca79